os_net_config/impl_nmstate.py

```diff
@@ -828,6 +830,21 @@ def generate_routes(self, interface_name):
                 logger.info(f'Prepare to remove route - {c_route}')
         return add_routes, del_routes
 
+    def managed_iprules(self, c_rule):
+        """
+        check if ip rule is manage by os-net.
+        only managed ip rule can be deleted, others will be ignored
+        """
+        if os.path.exists(MNG_RULES):
+            mng_iprules = ast.literal_eval(common.get_file_data(MNG_RULES))
+            for iprule in mng_iprules:
+                if c_rule['ip-from'] == iprule['ip-from'] and \
```

Comment:
Perform a strict match of all the options. Only options like metric shall be ignored